modules/overlays/custom-packages.nix

```diff
@@ -0,0 +1,55 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# This overlay is for custom packages - new packages, like Gala, or
+# fixed/adjusted packages from nixpkgs
+# The overlay might be used as an example and starting point for
+# any other overlays.
+#
+# !!!!!!! HINT !!!!!!!!
+# Use final/prev pair in your overlays instead of other variations
+# since it looks more logical:
+# previous (unmodified) package vs final (finalazed, adjusted) package.
+#
+# !!!!!!! HINT !!!!!!!!
+# Use deps[X][Y] variations instead of juggling dependencies between
+# nativeBuildInputs and buildInputs where possible.
+# It makes things clear and robust.
+#
+{
+  lib,
+  pkgs,
+  ...
+}: {
+  nixpkgs.overlays = [
+    (final: prev: {
+      # TODO: Remove this override if/when the fix is upstreamed.
+      # Adding missing dependencies for pipewire
+      pipewire = prev.pipewire.overrideAttrs (prevAttrs: {
+        nativeBuildInputs = prevAttrs.nativeBuildInputs ++ [pkgs.glib];
+        depsBuildBuild = [pkgs.gettext];
+      });
+      # TODO: Remove this override if/when the fix is upstreamed.
+      # Disabling colord dependency for weston. Colord has argyllcms as
+      # a dependency, and this package is not cross-compilable.
```

Comment:
For me the `argyllcms` seems to cross-compile just fine. I tested 22.11, unstable and master:

`$ nix build github:NixOS/nixpkgs/nixos-22.11#legacyPackages.x86_64-linux.pkgsCross.aarch64-multiplatform.argyllcms`

`$ nix build github:NixOS/nixpkgs/nixos-unstable#legacyPackages.x86_64-linux.pkgsCross.aarch64-multiplatform.argyllcms`

`$ nix build github:NixOS/nixpkgs/master#legacyPackages.x86_64-linux.pkgsCross.aarch64-multiplatform.argyllcms`

Reply:
Ok, seems that I missed something. I'll fix comments accordingly.

Reply:
Hmm, I still can't cross-compile `argyllcms`: 
```
nix-repl> :b pkgsCross.aarch64-multiplatform.argyllcms
error: builder for '/nix/store/2hzdi78k5fgz08j5344davpnx5wjkcrf-argyllcms-aarch64-unknown-linux-gnu-2.3.1.drv' failed with exit code 2;
       last 10 log lines:
       > imdi/imdi_make: line 2: syntax error: unexpected "("
       >
       > 	imdi/imdi_make -d imdi
       >
       > ...failed GenFileND1 imdi/imdi_k.h ...
       > Chmod1 yajl/json_verify
       > Chmod1 yajl/yajl_test
       > ...failed updating 1 target(s)...
       > ...updated 219 target(s)...
       > make: *** [Makefile:4: all] Error 1
       For full logs, run 'nix log /nix/store/2hzdi78k5fgz08j5344davpnx5wjkcrf-argyllcms-aarch64-unknown-linux-gnu-2.3.1.drv'.
 ```
 Frankly, I do not understand how `argyllcms` cross-compiles in your case since it's builder uses the single compiler for compiling itself and it's build-time dependencies (`imdi_make` in this case) which it then tries to run on buildHost.